### PR TITLE
Add Cursor.sh Doc extension

### PR DESCRIPTION
**Pull request by degouville**
## Description

Reach and search the [Cursor Documentation](https://www.cursor.sh/) in an instant right from Raycast.

## Screencast
sc1

## Checklist

- [x] I read the [extension guidelines](https://developers.raycast.com/basics/prepare-an-extension-for-store)
- [x] I read the [documentation about publishing](https://developers.raycast.com/basics/publish-an-extension)
- [x] I ran `npm run build` and [tested this distribution build in Raycast](https://developers.raycast.com/basics/prepare-an-extension-for-store#metadata-and-configuration)
- [x] I checked that files in the `assets` folder are used by the extension itself
- [x] I checked that assets used by the `README` are placed outside of the `metadata` folder

### CODE REVIEW

File: extensions/cursor/src/index.tsx [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+            icon={docsItem.icon ?? "cursor-icon.png"}
+            actions={
+              <ActionPanel>
+                <Action.OpenInBrowser
+                  title="Open the Cursor Documentation"
+                  icon={"cursor-icon.png"}
+                  url={docsItem.url || ""}
+                />
+                <Action onAction={revalidate} title="Reload Items" icon={Icon.Download} />
+                <Action onAction={openExtensionPreferences} title="Open Extension Preferences" icon={Icon.Gear} />
```
Comment (pernielsentikaer) [inline extensions/cursor/src/index.tsx:35]: I'm not sure if it makes sense to show preferences if no preferences are available

File: extensions/cursor/src/index.tsx [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+          {search.length > 0 && (
+            <List.Section>
+              <List.Item
+                icon={"cursor-icon.png"}
+                title={`Search '${search}' in the Cursor Documentation`}
+                actions={
+                  <ActionPanel>
+                    <OpenSearchInBrowserAction search={search ?? ""} />
+                    <Action onAction={revalidate} title="Reload Items" icon={Icon.Download} />
+                    <Action onAction={openExtensionPreferences} title="Open Extension Preferences" icon={Icon.Gear} />
```
Comment (pernielsentikaer) [inline extensions/cursor/src/index.tsx:52]: I'm not sure if it makes sense to show preferences if no preferences are available

**Review by pernielsentikaer — CHANGES_REQUESTED** [review]
Hi 👋 

Thanks for your contribution 💪 

I have now tested your extension, and I have some feedback ready for you:

- cursor.jpg is not in use, can we remove it

- Could you add at least one [screenshot](https://developers.raycast.com/basics/prepare-an-extension-for-store#screenshots) of the extension using our tool

I'm looking forward to testing this extension again 🔥 

Request a new review when you are ready. Feel free to contact me here or at [Slack](https://www.raycast.com/community) if you have any questions.

File: extensions/cursor/src/index.tsx (hunk 2 above)
Comment (degouville) [reply extensions/cursor/src/index.tsx]: ✅

**Comment by degouville** [comment]
> * Could you add at least one [screenshot](https://developers.raycast.com/basics/prepare-an-extension-for-store#screenshots) of the extension using our tool

Hi @pernielsentikaer, Thanks for your time and kind feedback. The raycast screenshot doesn't work for me. I got nothing on the metadata folder or clipboard. Can you help me on that please ?

**Comment by degouville** [comment]
Thanks for the screenshot @pernielsentikaer 👌🏾

LGTM✅

**Comment by degouville** [comment]
Thanks @pernielsentikaer ❣️

Have a good day ☀️

**Pull request merged**